gems/aws-sdk-core/lib/aws-sdk-core/xml/parser/frame.rb

```diff
@@ -266,7 +262,7 @@ def result
 
       class BlobFrame < Frame
         def result
-          @text.empty? ? nil : Base64.decode64(@text.join)
+          @text.empty? ? '' : Base64.decode64(@text.join)
```

Comment:
Re-examining this .... will this be a breaking behavior for customers? 
When they expect a nil value instead of empty string?

This adjustment was made due to this test case: https://github.com/smithy-lang/smithy/blob/09a61e2165b6d970987b4f035ded20b3c1ff2962/smithy-aws-protocol-tests/model/awsQuery/xml-structs.smithy#L183-L204

Reply:
Maybe? These methods would no longer exist on it. to_h might be of concern if it were called in some nested way (have to ensure that still works). But it's probably fine? I would agree with the test that if <data> tag didn't exist, it should be nil, but it's explicit and empty. I'd say it's fine and we could patch it if it's a problem.

```
irb(main):001> nil.methods - ''.methods
=> [:to_h, :&, :to_a, :rationalize, :|, :^]
```

Reply:
I think `to_h` in general should be okay on the containing structure (since to_h in general should be handling nils in other fields already).  I agree - I think this should be reasonable our other general behavior around nils.